python/dispatcher/dispatcher.py

```diff
@@ -0,0 +1,119 @@
+# -------------------------------------------------------------------------
+#
+# Copyright (c) 2024 General Motors GTO LLC
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# SPDX-FileType: SOURCE
+# SPDX-FileCopyrightText: 2024 General Motors GTO LLC
+# SPDX-License-Identifier: Apache-2.0
+#
+# -------------------------------------------------------------------------
+
+
+import socket
+from typing import Dict
+from threading import Thread
+from threading import Lock
+import uuid
+import logging 
+import sys
+
+
+PORT = 44444
+IP = "127.0.0.1" 
+ADDR = (IP, PORT)
+
+logging.basicConfig(format='%(asctime)s %(message)s')
+# create logger
+logger = logging.getLogger('simple_example')
+logger.setLevel(logging.INFO)
+
+
+# Holds all client connections
+clients: Dict[str, socket.socket] = {}
+# Create the shared lock
+lock = Lock()
+
+def close_socket(client_id: str, locket: Lock):
+    with locket:
+        clients[client_id].close()
+        del clients[client_id]
+
+def handle_client(clientsocket: socket.socket, client_id: str, locket: Lock):
```

Comment:
```python 
def handle_client(client_socket: socket.socket, client_id: str):
    while True:
        try:
            recv_data: bytes = client_socket.recv(32767)

            if not recv_data:
                continue

            logger.info(f"received data: {recv_data}")

            with child_lock:
                for peer_id, peer_socket in clients.items():
                    if peer_id == client_id:
                        continue

                    try:
                        peer_socket.sendall(recv_data)

                    except ConnectionAbortedError as e:
                        logger.error(f"Peer socket exception: {e}")
                        close_socket(peer_id)

        except ConnectionAbortedError as e:
            logger.error(f"Client socket receive exception: {e}")
            close_socket(client_id)
            break

```
This version ensures thread safety by using a lock (child_lock) to control access to shared resources (clients dictionary) and It uses sendall() instead of send() to ensure all data is sent reliably.